lib/ansible/modules/cloud/amazon/aws_config.py

```diff
@@ -0,0 +1,697 @@
+            params['recordingGroup'] = {}
+            if module.params.get('recording_group').get('all_supported') is not None:
+                params['recordingGroup'].update({
+                    'allSupported': module.params.get('recording_group').get('all_supported')
+                })
+            if module.params.get('recording_group').get('include_global_types') is not None:
+                params['recordingGroup'].update({
+                    'includeGlobalResourceTypes': module.params.get('recording_group').get('include_global_types')
+                })
+            if module.params.get('recording_group').get('resource_types'):
+                params['recordingGroup'].update({
+                    'resourceTypes': module.params.get('recording_group').get('resource_types')
+                })
+
+    if resource_type == 'aggregation_authorization':
+        params = {}
+        if module.params.get('authorized_account_id'):
+            params['AuthorizedAccountId'] = module.params.get('authorized_account_id')
+        if module.params.get('authorized_aws_region'):
+            params['AuthorizedAwsRegion'] = module.params.get('authorized_aws_region')
+
+    if resource_type == 'configuration_aggregator':
+        params = {}
+        if module.params.get('name'):
+            params['ConfigurationAggregatorName'] = module.params.get('name')
+        if module.params.get('account_sources'):
+            params['AccountAggregationSources'] = []
+            for i in module.params.get('account_sources'):
+                tmp_dict = {}
+                if i.get('account_ids'):
+                    tmp_dict['AccountIds'] = i.get('account_ids')
+                if i.get('aws_regions'):
+                    tmp_dict['AwsRegions'] = i.get('aws_regions')
+                if i.get('all_aws_regions') is not None:
+                    tmp_dict['AllAwsRegions'] = i.get('all_aws_regions')
+                params['AccountAggregationSources'].append(tmp_dict)
+        if module.params.get('organization_source'):
+            params['OrganizationAggregationSource'] = {}
+            if module.params.get('organization_source').get('role_arn'):
+                params['OrganizationAggregationSource'].update({
+                    'RoleArn': module.params.get('organization_source').get('role_arn')
+                })
+            if module.params.get('organization_source').get('aws_regions'):
+                params['OrganizationAggregationSource'].update({
+                    'AwsRegions': module.params.get('organization_source').get('aws_regions')
+                })
+            if module.params.get('organization_source').get('all_aws_regions') is not None:
+                params['OrganizationAggregationSourcep'].update({
+                    'AllAwsRegions': module.params.get('organization_source').get('all_aws_regions')
+                })
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
```

Comment:
These two lines (678 and 679) can be replaced with `client = module.client('config')` as this AWS connection discovery is now part of AnsibleAWSModule.